lib/cluster.rb

```diff
@@ -14,19 +14,26 @@
 # - htitems
 # - commitments
 class Cluster
-  attr_reader :ocns
+  attr_reader :ocns, :id
 
   def self.db
     Services.holdings_db
   end
 
-  def initialize(ocns: [])
+  def initialize(id: nil, ocns: [])
+    @id = id
     @ocns = ocns
   end
 
   def self.find(id:)
     ocns = db[:cluster_ocns].select(:ocn).where(cluster_id: id).map(:ocn)
-    new(ocns: ocns)
+    new(id: id, ocns: ocns)
+  end
+
+  def self.for_ocns(ocns)
```

Comment:
I found this a little tricky to parse, esp without clarifying comments.
Maybe split it apart into `<getting the rows>` and `<making cluster objects based on found rows>`

```ruby
rows = db[:cluster_ocns].select(:cluster_id).distinct.where(ocn: ocns)
rows.map do |row|
  find(id: row[:cluster_id])
end
```

Reply:
Let's discuss -- `db[:cluster_ocns].select(:cluster_id).distinct.where(ocn: ocns)` is essentially a glorified SQL statement, and it doesn't execute until you call `map` on it. I also changed these to enumerators here: https://github.com/hathitrust/holdings-backend/commit/328f970d9d876d3ba797b77d8d0f05178e5df746 

I think there will be a fair amount of this kind of stuff, so it's worth finding a pattern we're happy with.